pkg/build/bazel/bazel.go

```diff
@@ -18,16 +18,43 @@ import (
 	"os"
 	"path"
 	"path/filepath"
+	"strconv"
 	"strings"
 
 	inner "github.com/bazelbuild/rules_go/go/tools/bazel"
+	inner_testutil "github.com/bazelbuild/rules_go/go/tools/bzltestutil"
 )
 
+// bazelTestEnvVar can be used to determine when running in the `bazel test`
+// environment.
+const bazelTestEnvVar = "BAZEL_TEST"
+
 // Return true iff this library was built with Bazel.
 func BuiltWithBazel() bool {
 	return true
 }
 
+// Return true iff called from a test run by Bazel.
+func TestByBazel() bool {
+	if bazelTestEnv, ok := os.LookupEnv(bazelTestEnvVar); ok {
+		if bazelTest, err := strconv.ParseBool(bazelTestEnv); err == nil {
+			return bazelTest
+		}
+	}
+
+	return false
+}
+
+// Return true iff called from a test using Bazel's Go test wrapping.
+// When enabled and running under `bazel test`, the entire test runs using a
+// Bazel-generated wrapper. This wrapper imports the test module, so any
+// import-time code will be run twice: once under the wrapper, and once by the
+// test binary itself. For more, info, see:
+// https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/rules.md#go_test
+func TestWrapped() bool {
+	return inner_testutil.ShouldWrap()
```

Comment:
This doesn't tell you that you're inside the wrapper, only whether the wrapper should be used.

Reply:
The function is confusingly named (the whole thing is confusing so picking a not-confusing name is hard), but it does have the desired behavior: it returns `true` if we're in the "exterior" wrapping process, and `false` if we're in the forked-off test process.